config/database.go

```diff
@@ -295,16 +291,7 @@ func (ds *DatabaseStore) RemoveFile(name string) error {
 
 // String returns the path to the database backing the config, masking the password.
 func (ds *DatabaseStore) String() string {
-	// Remove @tcp and the parentheses from the host and parse the rest as a URL
-	u, err := url.Parse(tcpStripper.ReplaceAllString(ds.originalDsn, `@$1`))
-	if err != nil {
-		return "(omitted due to error parsing the DSN)"
-	}
-
-	// Strip out the password to avoid leaking in logs.
-	u.User = url.User(u.User.Username())
-
-	return u.String()
+	return stripPassword(ds.originalDsn)
```

Comment:
I think here we could leverage the fact that `ds.dataSourceName` is already stripped of the database driver schema and at the same time keep that check in only one place while avoiding the repetition in `stripPassword()`.

Also I think `String()` should return the original DSN if `stripPassword()` fails because not finding a password doesn't necessary mean that the whole DSN is not valid.

Reply:
> I think here we could leverage the fact that `ds.dataSourceName` is already stripped of the database driver schema and at the same time keep that check in only one place while avoiding the repetition in `stripPassword()`.

Not really. Because for `postgres` `parseDSN()` does not strip the schema https://github.com/mattermost/mattermost-server/blob/2158dadc4bf8e85a4af4e23dc6b01123a26df743/config/database.go#L114-L115.

> Also I think `String()` should return the original DSN if `stripPassword()` fails because not finding a password doesn't necessary mean that the whole DSN is not valid.

But that means we would print the password, if we fail to parse it, which we really want to avoid, right? I would rather be safe then sorry here.

Reply:
> 
> Not really. Because for `postgres` `parseDSN()` does not strip the schema
> 
> https://github.com/mattermost/mattermost-server/blob/2158dadc4bf8e85a4af4e23dc6b01123a26df743/config/database.go#L114-L115
> 

Right but we could at least use `ds.driverName` to build the prefix in `stripPassword`.

> But that means we would print the password, if we fail to parse it, which we really want to avoid, right? I would rather be safe then sorry here.

If the parsing fails it should mean there's no password. If you are suggesting the password is in the wrong place it will get out either way.
The problem with returning that error string is that any perfectly valid DSN without password will fail.

Reply:
> Right but we could at least use `ds.driverName` to build the prefix in `stripPassword`.

Yeah we can. Applied this change. 0/5 if this is really needed.


> > But that means we would print the password, if we fail to parse it, which we really want to avoid, right? I would rather be safe then sorry here.
> 
> If the parsing fails it should mean there's no password. If you are suggesting the password is in the wrong place it will get out either way.
> The problem with returning that error string is that any perfectly valid DSN without password will fail.

I've changed the function to be idempotent. This way it won't fail, if there is no password set. Does this solve your concern?

Reply:
> Yeah we can. Applied this change. 0/5 if this is really needed.

Great! Of course it's not really needed but I feel it's nicer to have so we don't need to update an additional function if we ever support more databases.

> If the parsing fails it should mean there's no password. If you are suggesting the password 
> I've changed the function to be idempotent. This way it won't fail, if there is no password set. Does this solve your concern?

My concern was that `stripPassword()` will fail with a valid DSN like `mysql://user@unix(/path/to/socket)/dbname`
I think it should strip a password if there is one, otherwise return the input.

Reply:
> My concern was that `stripPassword()` will fail with a valid DSN like `mysql://user@unix(/path/to/socket)/dbname`
> I think it should strip a password if there is one, otherwise return the input.

Fixed this case